trace/trace.go

```diff
@@ -28,29 +28,43 @@ import (
 type traceStep struct {
 	stepTime time.Time
 	msg      string
+	msgArgs  []interface{}
 }
 
 // Trace keeps track of a set of "steps" and allows us to log a specific
 // step if it took longer than its share of the total allowed time
 type Trace struct {
 	name      string
+	msg       string
+	msgArgs   []interface{}
 	startTime time.Time
 	steps     []traceStep
 }
 
 // New creates a Trace with the specified name
 func New(name string) *Trace {
-	return &Trace{name, time.Now(), nil}
+	return &Trace{name: name, startTime: time.Now()}
+}
+
+// Newf creates a Trace with the specified name and formatted msg
+func Newf(name string, msg string, args ...interface{}) *Trace {
+	return &Trace{name: name, startTime: time.Now()}
```

Comment:
Should this be setting `msgArgs`?

Reply:
You're right. I've replaced `Newf` with a optional `WithDetails` function which hopefully is clearer. I've fixed this along with that change.